nixos/modules/services/network-filesystems/openafs-server.nix

```diff
@@ -0,0 +1,119 @@
+  netInfo = if (cfg.advertisedAddresses != []) then
+    pkgs.writeText "NetInfo" ((concatStringsSep ''
+
+      f '' cfg.advertisedAddresses) + ''
+
+      '')
+    else "";
+  afsConfig = pkgs.runCommand "afsconfig" { } ''
+    mkdir -p $out
+    echo ${cfg.cellName} > $out/ThisCell
+    cp ${cellServDB} $out/CellServDB
+    cp ${cfg.bosConfig} $out/BosConfig
+    printf "${concatStringsSep "\n" cfg.adminUsers }" >$out/UserList
+    chmod 600 $out/UserList
+  '';
+  cellServDB = pkgs.runCommand "CellServDB" { } ''
+    printf ">${cfg.cellName}\n" > $out
+    printf "${concatStringsSep "\n" (map ({ address, hostname }: concatStringsSep " \#" [ address hostname ]) cfg.cellServers)}\n" >> $out
+  '';
+  bosConfig = pkgs.writeText "BosConfig" ''
+    restarttime 16 0 0 0 0
+    checkbintime 3 0 5 0 0
+    bnode simple vlserver 1
+    parm ${openafsPkgs}/libexec/openafs/vlserver
+    end
+    bnode simple ptserver 1
+    parm ${openafsPkgs}/libexec/openafs/ptserver -db /var/openafs/prdb -restricted
+    end
+    bnode dafs dafs 1
+    parm ${openafsPkgs}/libexec/openafs/dafileserver -vattachpar 128 -vhashsize 11 -cb 1000000
+    parm ${openafsPkgs}/libexec/openafs/davolserver
+    parm ${openafsPkgs}/libexec/openafs/salvageserver
+    parm ${openafsPkgs}/libexec/openafs/dasalvager
+    end
+  '';
+in
+{
+  options = {
+
+    services.openafsServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the OpenAFS server.";
+      };
+
+      adminUsers = mkOption {
+        type = types.listOf types.string;
+        default = [];
+        description = "List of OpenAFS users that can administer this server.";
+      };
+
+      advertisedAddresses = mkOption {
+        default = [];
+        description = "List of IP addresses this server is advertised under. See NetInfo(5)";
+      };
+      bosConfig = mkOption {
+        type = types.package;
+        default = bosConfig;
+        description = "The server configuration file specifying the started servers and options";
+      };
+
+      cellName = mkOption {
+        default = "grand.central.org";
+        description = "Server's cell name.";
+      };
+
+      cellServers = mkOption {
+        default = [];
+        description = "IP addresses of database servers that serve this cell.";
+        example = literalExample ''
+          [ { address = "10.0.0.3"; hostname = "afsdb.example.org"; } ]
+        '';
+      };
+
+      openafsPkgs = mkOption {
+        default = config.boot.kernelPackages.openafs;
+        description = "The package to use for the openafs servers";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ openafsPkgs ];
+
+    system.activationScripts.openafsServer = stringAfter
+    [ (if config.services.openafsClient.enable then "openafs" else "etc") ] ''
+      mkdir -p /etc/openafs/server
+      cp -r ${afsConfig}/* /etc/openafs/server #*/
```

Comment:
Is the comment at the end a left-over? Would it be possible to create a symlink by assigning the path of the config files to `environment.etc."openafs/server".source` instead of copying them at start up time?